cpp/util/export_Vector.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2010-2024 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#ifndef EXPORT_VECTOR_H
+#define EXPORT_VECTOR_H
+
+#include <nanobind/nanobind.h>
+#include <nanobind/ndarray.h>
+#include <string>
+#include <vector>
+
+#include "VectorMath.h"
+
+namespace freud { namespace util {
+
+namespace wrap {
+
+template<typename T> nanobind::ndarray<nanobind::numpy, const T> vectorToNumpyArray(nanobind::object self)
+{
+    std::vector<T>* self_cpp = nanobind::cast<std::vector<T>*>(self);
+    auto dims = {self_cpp->size()};
+    auto ndim = 1;
+    auto data_ptr = self_cpp->data();
+    return nanobind::ndarray<nanobind::numpy, const T>((void*) data_ptr, dims, self);
```

Comment:
This does not work. `self` here is the `std::vector` itself. The vector is actually held by a `PeriodicBuffer` (or another class instance). We can either
1) Make a copy of the data here (removes zero-copy performance advantage).
or 
2) make this a template method that `PeriodicBuffer` (and other classes) can use so that the `self` object is an instance of the holder.

The problem with option 2 is that the user can now invalidate arrays:
```
obj.compute()
array = obj.result
array.sum() # valid
obj.compute() # resizes internal array
array.sum() # seg fault
```

If we need zero-copy performance we need some kind of holder to keep these copies alive. That could be `ManagedArray`, but it doesn't have to be. We should test if it works, but I think holding `std::vector` in a `std::shared_ptr` would fix the seg fault. This would require minimal modification to the classes that use `std::vector`.

Reply:
Yes, it seems I overlooked this detail. Ideally, the C++ classes could use a `shared_ptr` to hold the vector, but I'm not sure how we would make that work given that the current pattern is to export the holder class to python and give it a `toNumpyArray` method.

Another option is to force usage of `ManagedArray` as the only data container we allow for array data on the C++ side. This is a bit restrictive, ideally there would be a more flexible option.